packages/flutter/test/material/theme_test.dart

```diff
@@ -231,4 +232,31 @@ void main() {
     expect(materials[0].color, green); // app scaffold
     expect(materials[1].color, green); // dialog scaffold
   });
+
+  testWidgets('IconThemes are applied', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      new MaterialApp(
+        theme: new ThemeData(iconTheme: const IconThemeData(color: Colors.green, size: 10.0)),
+        home: new Icon(Icons.computer),
+      )
+    );
+
+    RenderParagraph glyphText = tester.renderObject(find.byType(RichText));
+
+    expect(glyphText.text.style.color, Colors.green);
+    expect(glyphText.text.style.fontSize, 10.0);
+
+    await tester.pumpWidget(
+      new MaterialApp(
+        theme: new ThemeData(iconTheme: const IconThemeData(color: Colors.orange, size: 20.0)),
+        home: new Icon(Icons.computer),
+      ),
+    );
+    await tester.pump(const Duration(milliseconds: 200)); // For the theme to transition over.
```

Comment:
might be worth checking that the icon theme is actually lerping by seeing if half-way through, it is equal to neither orange nor green